axelor-purchase/src/main/resources/views/PurchaseOrderLine.xml

```diff
@@ -334,8 +334,12 @@
 	<!-- ACTION ATTRS -->
 	
 	<action-attrs name="action-purchase-order-line-attrs-product-domain">
-		<attribute name="domain" for="product"
-		expr="eval: isFilterOnSupplier ? &quot; self.isModel = false and self.expense = false and (self.endDate = null or self.endDate &gt; :__date__) and self.purchasable = true and self.id IN (SELECT product.id FROM SupplierCatalog WHERE supplierPartner.id = ${_parent?.supplierPartner?.id}) &quot; : &quot; self.isModel = false and self.expense = false and (self.endDate = null or self.endDate &gt; :__date__) and self.purchasable = true &quot;" />
+		<attribute name="domain" for="product" if="__config__.app.getApp('purchase').getManageSupplierCatalog()"
+		expr="eval: isFilterOnSupplier 
+		? &quot; self.isModel = false and self.expense = false and (self.endDate = null or self.endDate &gt; :__date__) and self.purchasable = true and self.id IN (SELECT product.id FROM SupplierCatalog WHERE supplierPartner.id = ${_parent?.supplierPartner?.id}) &quot; 
+		: &quot; self.isModel = false and self.expense = false and (self.endDate = null or self.endDate &gt; :__date__) and self.purchasable = true &quot;" />
+		<attribute name="domain" for="product" if="!__config__.app.getApp('purchase').getManageSupplierCatalog()"
+		expr="eval: &quot; self.isModel = false and self.expense = false and (self.endDate = null or self.endDate &gt; :__date__) and self.purchasable = true &quot;" />
 	</action-attrs>
```

Comment:
!isFilterOnSupplier or !__config__.app.getApp('purchase').getManageSupplierCatalog()   both are same condition, should be done in one attribute.